lib/resty/websocket/client.lua

```diff
@@ -134,11 +134,18 @@ function _M.connect(self, uri, opts)
             origin_header = "\r\nOrigin: " .. origin
         end
 
-        local pool = opts.pool
-        if pool then
-            sock_opts = { pool = pool }
+        if opts.pool then
+            sock_opts.pool = opts.pool
+        end
+        --pool_size specify the size of the connection pool. If omitted and no backlog option was provided, no pool will be created.
+        if opts.pool_size then
+            sock_opts.pool_size = opts.pool_size
+        end
+        if opts.backlog then
+            sock_opts.backlog = opts.backlog
         end
 
+
```

Comment:
FYI this extra line is also in upstream, so I kept it

https://github.com/openresty/lua-resty-websocket/blob/fbd01942507b1cfa0cbbaa792c3ccd46a3d66316/lib/resty/websocket/client.lua#L133-L134